module/solutions/helper.js

```diff
@@ -694,4 +745,334 @@ module.exports = class SolutionsHelper {
+     * @method
+     * @name solutionAndcriteriaDetails
+     * @param {String} - solutionId 
+     * @param {Boolean} [returnQuestion = false] - return criteria question or not 
+     * @returns {Object} - Criteria and questions information.
+     */
+
+    static solutionAndcriteriaDetails(
+      solutionId,
+      returnQuestion = false
+    ) {
+      return new Promise(async (resolve, reject) => {
+        try {
+
+          let solutionData = 
+          await this.solutionDocuments(
+              {
+                _id : solutionId
+              }, [
+                "creator",
+                "description",
+                "themes",
+                "type",
+                "evidenceMethods",
+                "linkTitle",
+                "linkUrl"
+              ]
+            );
+
+            if( !solutionData[0] ) {
+                throw {
+                    status : httpStatusCode.bad_request.status,
+                    message : messageConstants.apiResponses.SOLUTION_NOT_FOUND
+                }
+            }
+
+            const criteriasId = gen.utils.getCriteriaIds(solutionData[0].themes);
+            
+            let allCriteriaDocument = await criteriaHelper.criteriaDocument(
+                {
+                    _id : { $in : criteriasId}
+                },[
+                  "evidences"
+                ]
+            );
+
+            if( !allCriteriaDocument.length > 0 ) {
+              throw {
+                  status : httpStatusCode.bad_request.status,
+                  message : messageConstants.apiResponses.CRITERIA_NOT_FOUND
+              }
+            }
+
+            let result = {
+              solution : solutionData[0],
+              criteriasId : criteriasId,
+              criteriaDocuments : allCriteriaDocument
+            }
+
+            if( returnQuestion ) {
+              
+              let questionIds = gen.utils.getAllQuestionId(allCriteriaDocument);
+            
+              const questionDocuments = await questionsHelper.questionDocument(
+                {
+                  _id: { $in: questionIds }
+                },
+                [
+                  "question"
+                ]
+              );
+
+              result["questionIds"] = questionIds;
+              result["questionDocuments"] = questionDocuments;
+            }
+  
+          return resolve(result);
+
+        } catch (error) {
+          return reject(error);
+        }
+      });
+    }
+
+     /**
+     * Create observation.
+     * @method
+     * @name createProgramAndSolutionFromTemplate -
+     * @param {Object} filteredData
+     * @param {String} userId - Logged in user id.
+     * @param {Object} program - program information
+     * @param {Object} solution - solution information
+     * @param {Boolean} [isAPrivateProgram = false] - created program is private or not     
+     * @returns {Object} Created solution and program
+     */
+
+    static createProgramAndSolutionFromTemplate(
```

Comment:
@aman-tunerlabs - I would suggest breaking this up with respective things being done in their modules. For .e.g. program creation should not be here but rather in programs module. The relevant observation module can first create solution using the solution helper and then map the solution to the program as a next step. Separate out the logic to be reusable later on.